Add Colored Icon Support - Inspired by #22

heliguy4599 opened the pull request:
In issue #22 the feature request to add colored icon support featuring Amy's Adwaita Color's icon pack. This PR adds the support for those to be enabled from your extension. This is acchieved by the following:
- A new settings schemea for this extension is made called "theme-icons" of type boolean
- The individual icon themes are stored as tar.gz files within the `resources/<selected color>` directory
- The selected accent color is then unzipped to the `~/.local/share/icons/` folder
- A Gio.Settings object is made to apply the `org.gnome.desktop.interface icon-theme: custom-icon-colors` icon theme
- If default or blue is selected, or if the Theme Icons switch or extension is turned off, that gsetting is set back to "Adwaita" returning the theme to default

I've tested this on GNOME 45 via openSUSE Tumbleweed.

I've matched your coding and naming styles so that it should be easy to follow along with what has changed. I'm willing to make any changes you deem necessary and really hope this feature can be added! Thank you for an amazing extension!

heliguy4599 committed:
Test commit

heliguy4599 committed:
Add Amy's icon support

AtiusAmy commented:
That sounds pretty sick! :3

AtiusAmy commented:
Just saying, but if this gets merged, Am I allowed to submit it to This week in GNOME?

dimitriskp22 commented:
hey @heliguy4599, thanks for the pr! :) I don't have access to my main laptop at the moment, but I'll try to test it as soon as I can. I didn't notice any major issues at first glance, I'll make a comment here when I'm able to test everything.

heliguy4599 commented:
Alright that's awesome! I look forward to hearing about it. And no problem for the PR, it was a fun task to work on :)

dimitriskprs commented on custom-accent-colors@demiskp/extension.js:
```diff
@@ -203,4 +209,34 @@ export default class CustomAccentColors extends Extension {
+                'custom-icon-colors.tar.gz'
+            );
+            let iconThemeTarText = iconThemeTar.get_path()
+            GLib.spawn_command_line_sync(
```
It's advisable to not have GNOME extensions spawn commands. I think it would be better to have the icon dirs uncompressed in the "resources" dir and simply copy them from there to the appropriate dir, as I've done with the shell themes.

heliguy4599 replied:
Ah I was wondering about that. Problem is I don't think I can copy a folder at once. I will certainly start work on a file by file copy system, but if you know any way to copy a folder or use a GLib / Gio way of moving or extracting files I would love to hear it, I just haven't ran across any.

dimitriskprs commented on custom-accent-colors@demiskp/prefs.js:
```diff
@@ -117,6 +117,19 @@ class CustomAccentColorsPrefsWidget extends Adw.PreferencesPage {
+        this.settings.bind('theme-icons', toggle, 'active', Gio.SettingsBindFlags.DEFAULT);
+        extraRow = new Adw.ActionRow({
+            title: 'Icon Theming',
+            subtitle: 'Color folder icons and the Files app icon',
```
Maybe the subtitle isn't necessary, I think the title is descriptive enough.

heliguy4599 replied:
Alright, I've removed that in the most recent commit in this.

dimitriskp22 commented:
apart from those 2 minor issues everything else is fine :)

heliguy4599 committed:
Update prefs.js

Remove the theme icon row's subtitle

heliguy4599 committed:
Change to use symlinks instead of tar extractions

heliguy4599 committed:
Add yellow, red, purple, and orange icon themes

heliguy4599 committed:
Add magenta, green, gray, and brown icon themes

heliguy4599 commented:
Alright, @dimitriskp22, I've removed that row subtitle but more importantly, I've changed my implementation so that as you request, the themes are stored as regular folders and a symlink is made that points to the currently selected accent color's matching theme. If the accent color changes, the symlink is deleted and a new one is made pointing to the newly selected accent color. And all achieved without spawning commands!

dimitriskp22 commented:
@heliguy4599 thanks! :) I'll test everything in a vm as soon as I can.

heliguy4599 commented:
Awesome! Can't wait!

dimitriskp22 commented:
@heliguy4599 I just tested everything and everything works great! :) thank you for the pr!

heliguy4599 commented:
Yay! And yeah no problem. Happy to help :D

AtiusAmy commented:
> I would like to know why the changes were reverted and are not available in the latest version

Maybe because i abandoned the project, im sorry but i really not have the energy for the Linux community rn, just not a fun place to be around...

demetrisk03 commented:
hey @antos07, the colored icons support never got into the extensions.gnome.org version of the extension because the filesize of the extension with all the extra icon packs was too big, so when the icon pack was discontinued I decided to remove the feature, an older version of the extension with the colored icons support is still in the repo in a separate branch but it's a lot of commits behind the main branch by now.
